x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/CcrRequests.java

```diff
@@ -24,8 +32,23 @@ public static ClusterStateRequest metaDataRequest(String leaderIndex) {
 
     public static PutMappingRequest putMappingRequest(String followerIndex, MappingMetaData mappingMetaData) {
         PutMappingRequest putMappingRequest = new PutMappingRequest(followerIndex);
+        putMappingRequest.origin("ccr");
         putMappingRequest.type(mappingMetaData.type());
         putMappingRequest.source(mappingMetaData.source().string(), XContentType.JSON);
         return putMappingRequest;
     }
+
+    public static final MappingRequestOriginValidator CCR_PUT_MAPPING_REQUEST_VALIDATOR = (request, state, indices) -> {
+        final List<Index> followingIndices = Arrays.stream(indices)
+            .filter(index -> {
+                final IndexMetaData indexMetaData = state.metaData().index(index);
+                return indexMetaData != null && CcrSettings.CCR_FOLLOWING_INDEX_SETTING.get(indexMetaData.getSettings());
+            }).collect(Collectors.toList());
+        if (followingIndices.isEmpty() == false && "ccr".equals(request.origin()) == false) {
+            return new IllegalStateException("can't put mapping to the following indices [" +
```

Comment:
Maybe throw an ElasticSearchStatus exception here with a 403 response code? `IllegalStateException` will result into a 500 response code and in this case a user is something that is not allowed.

Reply:
++. I pushed https://github.com/elastic/elasticsearch/pull/37675/commits/e83679dae5415f575c6dce8e0e02995ce599ef29.